contracts/Pool.sol

```diff
@@ -127,21 +137,19 @@ contract Pool {
         uint256 collected = 0;
         int256 lastFundsPerCycle = receiver.lastFundsPerCycle;
         for (; collectedCycle <= currFinishedCycle; collectedCycle++) {
-            int256 delta = receiver.amtDeltas[collectedCycle];
-            if (delta != 0) {
-                delete receiver.amtDeltas[collectedCycle];
-                lastFundsPerCycle += delta;
-            }
```

Comment:
I think it makes sense to keep this. If `delta == 0` and we delete the storage then this will cost gas. We only receive a refund from `delete` if the storage slot was non-zero.

Reply:
But now `delta` comes from both `receiver.amtDeltas[collectedCycle - 1].nextCycle` and `receiver.amtDeltas[collectedCycle].thisCycle;`. To decide whether or not to `delete receiver.amtDeltas[collectedCycle - 1];` we would need to either read `receiver.amtDeltas[collectedCycle - 1].thisCycle` too to confirm that it was non-zero, which costs 200 gas, exactly as much as setting storage from 0 to 0. I think that this is the cheapest option possible, any additional read would immediately obliterate any possible savings.